src/main/java/com/gachigang/ontherun/service/DepartmentService.java

```diff
@@ -1,25 +1,43 @@
 package com.gachigang.ontherun.service;
 
 import com.gachigang.ontherun.common.exception.NotFoundException;
+import com.gachigang.ontherun.common.mapper.DepartmentMapperImpl;
+import com.gachigang.ontherun.model.dto.DepartmentDto;
 import com.gachigang.ontherun.persistence.entity.Department;
+import com.gachigang.ontherun.persistence.repository.BusinessRepository;
 import com.gachigang.ontherun.persistence.repository.DepartmentRepository;
 import com.gachigang.ontherun.persistence.repository.UserRepository;
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.HashSet;
+import java.util.Set;
 
 @Service
 @RequiredArgsConstructor
 public class DepartmentService {
 
     private final DepartmentRepository departmentRepository;
     private final UserRepository userRepository;
+    private final BusinessRepository businessRepository;
 
     @Transactional(readOnly = true)
     public Department getDepartmentByUserId(@NonNull final Long id) {
         return departmentRepository.findDepartmentByUsersContains(userRepository.findById(id)
                 .orElseThrow(NotFoundException::new));
     }
+
+    @Transactional(readOnly = true)
+    public Set<DepartmentDto> getDepartmentsByBusinessId(@NonNull Long businessId) {
+        Set<Department> departments = departmentRepository.findDepartmentsByBusiness(businessRepository.findById(businessId)
```

Comment:
Hmmmmm as far as I remember, all endoints witch return Collection of objects shoud return an empty collection instead of throwing error. It is way more efficient and handy to use from angular perspective. Even, repositpry layer should return emtry set if there is no departments